fr/lecons/introduction-a-la-stylometrie-avec-python.md

```diff
@@ -0,0 +1,663 @@
+
+Si nous ne disposions d'aucune information supplémentaire, il faudrait se résoudre à conclure (sans beaucoup de confiance) que les articles contestés sont probablement l'oeuvre de Madison. Mais heureusement, la stylométrie a beaucoup progressé depuis l'époque de Mendenhall.
+
+# Deuxième test stylométrique : la méthode du khi carré de Kilgariff
+
+En 2001, Adam Kilgariff[^15] a proposé d'utiliser la statistique du khi carré (également nommée "khi deux") pour déterminer l'auteur d'un texte anonyme. Les lecteurs familiers avec les méthodes quantitatives savent peut-être que le khi carré est parfois utilisé pour déterminer si un ensemble d'observations, par exemple les intentions de vote énoncées par les électeurs lors d'un sondage, suivent une [loi de probabilité](https://fr.wikipedia.org/wiki/Loi_de_probabilit%C3%A9) particulière. Ce n'est pas ce que nous cherchons ici. Nous utiliserons plutôt la statistique pour mesurer la "distance" entre les vocabulaires employés dans deux ensembles de texte. Plus ces vocabulaires sont similaires, plus il est probable que les ensembles de textes sous-jacents soient l'oeuvre de la même personne. (Cela suppose que le vocabulaire d'une personne et ses habitudes d'usage des mots soient relativement constants.)
+
+Voici comment appliquer la statistique du khi carré à notre problème d'identification :
+
+* Prenons les corpus associés à deux auteurs.
+* Fusionnons-les en un seul corpus combiné de plus grande taille.
+* Comptons les occurrences de tous les mots qui apparaissent dans ce corpus combiné.
+* Conservons les [n](https://fr.wikipedia.org/wiki/%C3%89chantillon_(statistiques)) mots les plus fréquents.
+* Calculons le nombre d'occurrences de ces `n` mots les plus fréquents que l'on se serait attendu à observer dans chacun des deux corpus originaux si ceux-ci avaient été rédigés par la même personne. Pour ce faire, il faut diviser le nombre d'occurrences observé dans le corpus combiné, proportionnellement aux tailles des deux corpus originaux.
+* Calculons un khi carré en additionnant, pour chacun des `n` mots les plus fréquents, _les carrés des différences entre les nombres d'occurrences observés dans les corpus des auteurs et les nombres d'occurrences attendus, divisés par les nombres d'occurrences attendus_. La figure 6 montre la formule de calcul du khi carré, où C(i) représente le nombre d'occurrences observé pour le mot 'i' tandis que E(i) représente le nombre d'occurrences attendu pour ce mot.
+
+{% include figure.html filename="stylometry-python-6.jpg" caption="Figure 6: Formule de calcul du khi carré." %}
```

Comment:
If @fdlaramee is comfortable writing it, another option is to write this formula in LaTeX and add `mathjax: true` to the YAML header, that way it will render as selectable text instead of an image. Just an option - if you're both happy using the jpg image then that's fine with me.